security/pkg/stsservice/tokenmanager/tokenmanager.go

```diff
@@ -90,6 +91,8 @@ func CreateTokenManager(tokenManagerType string, config Config) security.TokenMa
 				projectInfo.Number, projectInfo.clusterURL, true); err == nil {
 				tm.plugin = p
 			}
+		} else {
+			log.Warnf("%v token manager specified but failed to ready GCP project information", GoogleTokenExchange)
```

Comment:
Should the failing to ready GCP project be treated as an error and return the error?

Reply:
I think we need a higher level decision. Either GCE metadata failing is an error and we exit, or its not and we gracefully recover.

Right now we do neither - if GCE metadata fails we just start having things break and never really recover.

Reply:
How about propagating the errors to higher levels and exit if necessary or retry a few times to recover?

Reply:
Its not quite that simple, we pull from metadata server in 5+ different places. So it would be a bigger change than this pr was intended, but seems like it could be the right solution.

@douglas-reid @bianpengyuan  any thoughts?

Reply:
the platform metadata stuff tries a few different ways to get the project information. we could add retries in the platform package, but that comes at the cost of (potentially) slowing things down when there are issues with the metadata server (or when not running on GCP).

Do we have another use case where we need to exit if a particular field of the platform metadata isn't populated?  Or can we start with some sort of a local solution here in `GetGCPProjectInfo()` ?

BTW, if we can confirm we are on GCP and we can't get project info, maybe that's sufficient to exit, but we'd need to ensure that off-GCP use cases weren't impacted by that decision.